File: Data Structure and Algorithm/Assignment 01/Program 09/prog9.py
#Write a C program to delete a range of data from a dynamic array.

import array as arr

a1 = arr.array('i')
n = int(input("Enter the number of inputs: "))
for i in range(n):
    val = int(input(f"Enter the element{i+1}: "))
    a1.append(val)

start = int(input("Enter the start range: "))
end = int(input("Enter the end range: "))
new_idx = end-start+1
for i in range(end+1,n):
    a1[start] = a1[i]
    start+=1

del a1[n-new_idx:]
print(a1)

File: Assignment 01/Program 09/prog9.c
// Write a C program to delete a range of data from a dynamic array.

#include<stdio.h>
#include<stdlib.h>

int main(){
    int n;
    printf("Enter the number of inputs: ");
    scanf("%d",&n);
    int *arr = (int*)malloc(n*sizeof(int));
    for (int i = 0; i < n; i++)
    {
        printf("Enter the element %d : ",i+1);
        scanf("%d",&arr[i]);
    }

    int start,end ;
    printf("Enter the ranges(start, end): ");
    scanf("%d %d",&start, &end);

    int newsize = n-(end-start+1);
    for (int i = end+1; i < n; i++)
    {
        arr[start++] = arr[i];
    }
    
    arr = (int*)realloc(arr,newsize*sizeof(int));

    for (int i = 0; i < newsize; i++)
    {
        printf("%d ",arr[i]);
    }    
    free(arr);
    return 0;
}